RELBASE: ./vendor/odoo/cc


# odoo/tools/config.py - [PATCH] [IMP] config: support config folders
https://github.com/xoe-labs/odoo-dev/compare/10.0...xoe-labs:10.0-NOPR-support-config-folder.patch

From dee689ea37e1100498232689fbf4fe84bfbd1a9c Mon Sep 17 00:00:00 2001
From: David Arnold <dar@xoe.solutions>
Date: Mon, 20 Aug 2018 13:00:16 -0500
Subject: [PATCH] [IMP] config: support config folders

This commit helps organizing the configuration file.
For example, if part of your configuration has an uncoupled lifecycle.

Signed-off-by: "David Arnold" <"dar@xoe.solutions">
---
 odoo/tools/config.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/odoo/tools/config.py b/odoo/tools/config.py
index be62f7c8b5a85..a59c82cdd4956 100644
--- a/odoo/tools/config.py
+++ b/odoo/tools/config.py
@@ -366,6 +366,10 @@ def die(cond, msg):

         self.rcfile = os.path.abspath(
             self.config_file or opt.config or os.environ.get('ODOO_RC') or os.environ.get('OPENERP_SERVER') or rcfilepath)
+        if os.path.isdir(self.rcfile):
+            self.rcfiles = sorted([os.path.join(self.rcfile, f) for f in os.listdir(self.rcfile) if os.path.isfile(os.path.join(self.rcfile, f))])
+        else:
+            self.rcfiles = [self.rcfile]
         self.load()

         # Verify that we want to log or not, if not the output will go to stdout
@@ -503,7 +507,7 @@ def _check_addons_path(self, option, opt, value, parser):
     def load(self):
         p = ConfigParser.ConfigParser()
         try:
-            p.read([self.rcfile])
+            p.read(self.rcfiles)
             for (name,value) in p.items('options'):
                 if value=='True' or value=='true':
                     value = True